.github/workflows/docker-presubmit.yml

```diff
@@ -41,4 +41,30 @@ jobs:
           tags: tensorflow/build:latest-${{ matrix.python-version }}
           cache-from: type=registry,ref=tensorflow/build:latest-${{ matrix.python-version }}
           cache-to: type=inline
-
+      -
+        name: Build containers CUDA
+        id: docker_build_cuda
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          context: ./tf_sig_build_dockerfiles
+          target: devel-cuda
+          build-args: |
+            PYTHON_VERSION=${{ matrix.python-version }}
+          tags: tensorflow/build:latest-${{ matrix.python-version }}-CUDA
+          cache-from: type=registry,ref=tensorflow/build:latest-${{ matrix.python-version }}-CUDA
+          cache-to: type=inline
+  compile:
+    needs: docker
+    runs-on: ubuntu-latest
+    container: tensorflow/build:latest-python3.9
```

Comment:
Please note that on this line we are using the image from Dockerhub to minimize the cache misses derived from env changes.